File: TECH-AGENT/app/api/v1/plans.py
"""Autonomous task planning endpoints (V15-02)."""

from __future__ import annotations

from typing import Optional

from fastapi import APIRouter, Depends, Query, Request

from app.common.api_response import build_page, success
from app.common.context import RequestContext, request_context_dep
from app.deps import get_plan_service
from app.plans.schemas import plan_to_dict
from app.plans.service import PlanService

router = APIRouter(tags=["plans"])


@router.post("/plans", summary="根据用户输入生成任务计划")
async def create_plan(
    body: dict,
    request: Request,
    ctx: RequestContext = Depends(request_context_dep),
    service: PlanService = Depends(get_plan_service),
) -> dict:
    from app.plans.schemas import CreatePlanRequest

    req = CreatePlanRequest(**body)
    plan = await service.create(ctx.tenant_id, req)
    return success(plan_to_dict(plan), trace_id=ctx.trace_id)


@router.get("/plans", summary="任务计划列表(分页)")
async def list_plans(
    request: Request,
    agentId: Optional[str] = Query(default=None),
    page: int = Query(default=1, ge=1),
    pageSize: int = Query(default=20, ge=1, le=100),
    ctx: RequestContext = Depends(request_context_dep),
    service: PlanService = Depends(get_plan_service),
) -> dict:
    items, total = await service.list(
        ctx.tenant_id,
        agent_id=agentId,
        page=page,
        page_size=pageSize,
    )
    paged = build_page(
        [plan_to_dict(p) for p in items],
        total=total,
        page=page,
        page_size=pageSize,
    )
    return success(paged, trace_id=ctx.trace_id)


@router.get("/plans/{plan_id}", summary="查询任务计划状态")
async def get_plan(
    plan_id: str,
    request: Request,
    ctx: RequestContext = Depends(request_context_dep),
    service: PlanService = Depends(get_plan_service),
) -> dict:
    plan = await service.get(ctx.tenant_id, plan_id)
    return success(plan_to_dict(plan), trace_id=ctx.trace_id)


@router.post(
    "/plans/{plan_id}/steps/{step_id}/approve",
    summary="批准计划步骤",
)
async def approve_step(
    plan_id: str,
    step_id: str,
    request: Request,
    ctx: RequestContext = Depends(request_context_dep),
    service: PlanService = Depends(get_plan_service),
) -> dict:
    plan = await service.approve_step(ctx.tenant_id, plan_id, step_id)
    return success(plan_to_dict(plan), trace_id=ctx.trace_id)


@router.post(
    "/plans/{plan_id}/steps/{step_id}/skip",
    summary="跳过计划步骤",
)
async def skip_step(
    plan_id: str,
    step_id: str,
    request: Request,
    ctx: RequestContext = Depends(request_context_dep),
    service: PlanService = Depends(get_plan_service),
) -> dict:
    plan = await service.skip_step(ctx.tenant_id, plan_id, step_id)
    return success(plan_to_dict(plan), trace_id=ctx.trace_id)


@router.post("/plans/{plan_id}/execute", summary="执行任务计划")
async def execute_plan(
    plan_id: str,
    request: Request,
    ctx: RequestContext = Depends(request_context_dep),
    service: PlanService = Depends(get_plan_service),
) -> dict:
    plan = await service.execute(ctx.tenant_id, plan_id)
    return success(plan_to_dict(plan), trace_id=ctx.trace_id)

File: metaplatform-frontend/apps/web/src/api/superai/plans.ts
import { createApiClient, apiPath } from '@mate/shared/api';

const client = createApiClient({ baseURL: apiPath('copilot', '') });
const data = <T>(resp: { data: T }): T => resp.data;
async function get<T>(url: string, params?: Record<string, unknown>): Promise<T> { return data(await client.get<T>(url, params ? { params } : undefined)); }
async function post<T>(url: string, body?: unknown): Promise<T> { return data(await client.post<T>(url, body)); }
async function put<T>(url: string, body?: unknown): Promise<T> { return data(await client.put<T>(url, body)); }
async function del<T>(url: string): Promise<T> { return data(await client.delete<T>(url)); }

import type { Plan, CreatePlanRequest } from './types';
const PLANS_BASE = '/plans';
export async function createPlan(req: CreatePlanRequest): Promise<Plan> {
  return post<Plan>(PLANS_BASE, req);
}
export async function getPlan(planId: string): Promise<Plan> {
  return get<Plan>(`${PLANS_BASE}/${planId}`);
}
export async function listPlans(params?: {
  agentId?: string;
  page?: number;
  pageSize?: number;
}): Promise<{ items: Plan[]; total: number; page: number; pageSize: number }> {
  return get(PLANS_BASE, params as Record<string, unknown> | undefined);
}
export async function approveStep(planId: string, stepId: string): Promise<Plan> {
  return post<Plan>(`${PLANS_BASE}/${planId}/steps/${stepId}/approve`);
}
export async function skipStep(planId: string, stepId: string): Promise<Plan> {
  return post<Plan>(`${PLANS_BASE}/${planId}/steps/${stepId}/skip`);
}
export async function executePlan(planId: string): Promise<Plan> {
  return post<Plan>(`${PLANS_BASE}/${planId}/execute`);
}
